persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/GeneMapper.xml

```diff
@@ -4,16 +4,115 @@
 <mapper namespace="org.cbioportal.persistence.mybatis.GeneMapper">
     <cache/>
 
-    <select id="getGeneListByHugoSymbols" resultType="org.cbioportal.model.Gene">
-        select ENTREZ_GENE_ID as entrezGeneId,
-        HUGO_GENE_SYMBOL as hugoGeneSymbol,
-        TYPE as type,
-        CYTOBAND as cytoband,
-        LENGTH as length 
-        from gene where HUGO_GENE_SYMBOL in
-                <foreach item="item" collection="hugo_gene_symbol" open="(" separator="," close=")">
-                    #{item}
-                </foreach>
+    <sql id="select">
+        gene.ENTREZ_GENE_ID AS "${prefix}entrezGeneId",
+        gene.HUGO_GENE_SYMBOL AS "${prefix}hugoGeneSymbol"
+        <if test="projection == 'SUMMARY' || projection == 'DETAILED'">
+            ,
+            gene.TYPE AS "${prefix}type",
+            gene.CYTOBAND AS "${prefix}cytoband",
+            gene.LENGTH AS "${prefix}length"
+        </if>
+    </sql>
+
+    <select id="getGenes" resultType="org.cbioportal.model.Gene">
+        SELECT
+        <include refid="select">
+            <property name="prefix" value=""/>
+        </include>
+        FROM gene
+        <if test="sortBy != null and projection != 'ID'">
+            ORDER BY ${sortBy} ${direction}
+        </if>
+        <if test="projection == 'ID'">
```

Comment:
In other endpoints do we order by the returned ID when using projection ID?  I think it is a good idea ... if we don't do it elsewhere, it would be good to carry it over to other endpoints. We could respect the direction of sort argument though .. here we hardcode it to be ASC (not a required change)